ocrd/ocrd/cli/workspace.py

```diff
@@ -154,15 +154,16 @@ def workspace_add_file(ctx, file_grp, file_id, mimetype, page_id, ignore, check_
 
     kwargs = {'fileGrp': file_grp, 'ID': file_id, 'mimetype': mimetype, 'pageId': page_id, 'force': force, 'ignore': ignore}
     log = getLogger('ocrd.cli.workspace.add')
+    log.debug("Adding '%s' (%s)", fname, kwargs)
     if not (fname.startswith('http://') or fname.startswith('https://')):
         if not fname.startswith(ctx.directory):
-            if exists(join(ctx.directory, fname)):
-                fname = join(ctx.directory, fname)
+            if exists(ctx.directory + '/' + fname):
+                fname = ctx.directory + '/' + fname
```

Comment:
Is this really worth giving up the abstraction of `os.path.join` – what about support for other path name conventions (like Windows)?

One could specifically intercept the problematic case (absolute + absolute)…

Reply:
That was just a quick fix, I'll implement it with pathlib.

Reply:
I decided to just check whether `fname` is absolute. Once (if ever) we don't need to support 3.5 anymore, we can refactor a lot of os.path/shutil code with pathlib but until then we better avoid mixing the two.